docs/spec.md

```diff
@@ -81,7 +81,7 @@ completely numeric to simplify generation of StableHLO programs.
 
 ```ebnf
 Type         ::= ValueType | NonValueType
-ValueType    ::= TensorType | TokenType | TupleType
+ValueType    ::= TensorType | QuantizedTensorType | TokenType | TupleType
```

Comment:
When doing another review of the PR, I noticed that we have a lot of "non-quantized tensor" type annotations.

This makes sense from the perspective of our speccing strategy - start small, make quantization opt in and expand on op-by-op basis. However, from the perspective of readers who aren't yet interested in quantization, the "non-quantized tensor" type annotations are just overhead that is sprinkled all over the spec.

I think a better approach for now would be to split non-quantized and quantized tensor types. This way type annotations become opt in ("tensor or quantized tensor") rather than opt out ("non-quantized tensor"), which better matches the current speccing strategy.

On a different note, the fact that per-axis quantization is encoded in the element type rather than the tensor type is not exactly amazing, so separating non-quantized and quantized tensor types may make sense from that perspective too. Let's not worry about this in the context of the current PR though.